### ISSUE TITLE
[Bug]: NC 26 - OCA\Text\Service\EncodingService::encodeToUtf8(): Argument #1 ($string) must be of type string, bool given

### ISSUE THREAD

**Issue by MikeDev101** [nextcloud/text#3980]

**Comment by MikeDev101** [comment]
Seems like re-installing Nextcloud Office and the Built-in CODE server seems to have done the trick. Close or no?
![image](https://user-images.githubusercontent.com/12957745/227238185-683d4aec-6e6c-4153-9e57-7b5e575ca4dc.png)

**Comment by juliushaertl** [comment]
I don't think installing the built-in code server is related. It looks like there was a temporary issue when reading the file, which might have caused the exception. We should handle this case more gracefully though, so let's keep this open.

**Comment by juliushaertl** [comment]
Note that getContent should just read the file from the filesystem which should just call fopen. Was there any log message before the one you pasted that could indicate why this failed?

**Comment by MikeDev101** [comment]
> Note that getContent should just read the file from the filesystem which should just call fopen. Was there any log message before the one you pasted that could indicate why this failed?

Other than various instances of user login failures, I noticed a similar error coming from the mail client:

![image](https://user-images.githubusercontent.com/12957745/227241893-2401a328-4706-4cb0-8494-dd6af3224db9.png)

Otherwise, no.

**Comment by NickCloudAT** [comment]
I have pretty much the same problem.. Since the Upgrade to 26 I cannot open ANY file from the Browser or APP anymore.
Not a txt, not a md not a office document, nothing :/

I also get the following error in logs:

`Exception: OCA\Text\Service\EncodingService::encodeToUtf8(): Argument #1 ($string) must be of type string, bool given, called in /var/www/nextcloud/apps/text/lib/Service/ApiService.php on line 279 in file '/var/www/nextcloud/apps/text/lib/Service/EncodingService.php' line 39`

I already tried various things (except tinkering with source files) > Updates etc pretty much..
I really depend on the feature to edit files in web, would be nice to know how to fix this.

**Comment by mic05** [comment]
I have exactly the same issue after upgrading to 26.0.1. I traced it down to the Common.php line 198:

https://github.com/nextcloud/server/blob/46459ae93fd2fe76a684569d802e7c8972098005/lib/private/Files/Storage/Common.php#LL198C44-L198C44

> public function file_get_contents($path) {
	$handle = $this->fopen($path, "r");
	if (!$handle) {
		return false;
	}
	$data = stream_get_contents($handle);
	fclose($handle);
	return $data;
}


Seems that the handle is `false` and therefore false is returned instead of the content of the file. Sadly, I'm no developer and couldn't trace it down further. 
Can someone tell me how to fix this or is this an error in nextcloud 26 or in text in general?

**Comment by juliushaertl** [comment]
Can you test if you can actually download the affected file through the file list? Might be worth to see if the file is still there in the data directory of your nextcloud instance.

**Comment by mic05** [comment]
> Can you test if you can actually download the affected file through the file list? Might be worth to see if the file is still there in the data directory of your nextcloud instance.

Yeah, all files can be downloaded via the file list just fine, only the text editor and collabora can't open the files.

**Comment by fanonwue** [comment]
Just wanted to chime in that I've encountered this problem as well after upgrading to NC 26. The `getContent()` function of the File object returns false for some reason, but downloading the file works perfectly fine.

**Comment by mic05** [comment]
@fanonwue Thanks for this! 
@juliushaertl Rescanning the files worked like a charm. No idea why.

**Comment by mic05** [comment]
@juliushaertl After a few days I cannot open the files again, same problem. Rescanning doesn't work now. What can I do?

**Comment by juliushaertl** [comment]
Any chance you can share the file if it is specific to one and it's content? Or a similarly broken file?

**Comment by juliushaertl** [comment]
Feel free to also share it by mail if you don't feel comfortable putting it here

**Comment by MikeDev101** [comment]
I gave up on this issue a while ago and reinstalled Nextcloud using NC AIO.

**Comment by mic05** [comment]
After upgrading to NC 26.0.2 the issue no longer persists. All files can be opened again.

**Comment by mejo-** [comment]
@fbrendel and @NickCloudAT are you still able to reproduce the issue with latest Nextcloud 26.0.2? If so, please report and I'll reopen the issue.

**Comment by NickCloudAT** [comment]
> @fbrendel and @NickCloudAT are you still able to reproduce the issue with latest Nextcloud 26.0.2? If so, please report and I'll reopen the issue.

Hey, no more errors. Works fine now.
